src/nodes.coffee

```diff
@@ -1770,11 +1802,86 @@ exports.Assign = class Assign extends Base
     answer = compiledName.concat @makeCode(" #{ @context or '=' } "), val
     # Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assignment_without_declaration,
     # if we’re destructuring without declaring, the destructuring assignment must be wrapped in parentheses.
-    if o.level > LEVEL_LIST or (isValue and @variable.base instanceof Obj and not @param)
+    if o.level > LEVEL_LIST or (isValue and @variable.base instanceof Obj and (not @param or @paramWithSplat))
       @wrapInParentheses answer
     else
       answer
 
+  # Check object destructuring variable for rest elements;
+  # can be removed once ES proposal hits Stage 4.
+  compileObjectDestruct: (o) ->
+    # Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assignment_without_declaration,
+    # if we’re destructuring without declaring, the destructuring assignment must be wrapped in parentheses.
+    # ({a, b} = obj)
+    # Helper function setScopeVar() declares vars 'a' and 'b' at the top of the current scope.
+    setScopeVar = (obj) ->
+      newVar = false
+      return if obj instanceof Assign and obj.value.base instanceof Obj
+      if obj instanceof Assign
+        if obj.value.base instanceof IdentifierLiteral
+          newVar = obj.value.base.compile o
+        else 
+          newVar = obj.variable.base.compile o
+      else
+        newVar = obj.compile o
+      o.scope.add(newVar, 'var', true) if newVar
+    # Helper function getPropVaues() returns object compiled property value.
+    # These values are then passed as argument to helper function objectWithoutKeys which is used to assign object value to the destructuring rest variable.
+    getPropValue = (obj) ->
+      wrapInQutes = (obj) -> (new Literal("'#{obj.compile o}'")).compile o
+      setScopeVar obj # Declare a variable in the scope.
+      if obj instanceof Assign
+        return obj.variable.compile o if obj.variable.base instanceof StringWithInterpolations or obj.variable.base instanceof StringLiteral
```

Comment:
I wonder if it would be safe for `StringWithInterpolations` to `extends StringLiteral`? I guess it doesn't really qualify as a 'literal'...

Out of scope for this PR, anyway.